Adding support for overloaded method names

nxtstep opened the pull request:
Make sure all methods are mockable/stubbed when method names are overloaded

nxtstep commented:
@TadeasKriz could you please check this PR?

TadeasKriz commented:
Hi, sorry for the long delay. I finally merged huge change and I'd like to ask you if you could try and rebase this PR on top of current master. Then I'd like to merge it. Thanks!

nxtstep commented:
@TadeasKriz My apologies for the long delay as well. Just updated and fixed the feature I wanted to support: stubbing/mocking Protocol extension functions that implement a protocol method and method name parameter overloading
